consutil/main.py

```diff
@@ -62,10 +67,56 @@ def clear(linenum):
 
 # 'connect' subcommand
 @consutil.command()
-@click.argument('linenum')
-def connect(linenum):
-    """Connect to switch via console device"""
-    click.echo("connect linenum")
+@click.argument('target')
+@click.option('--devicename', '-d', is_flag=True, help="connect by name - if flag is set, interpret linenum as device name instead")
+def connect(target, devicename):
+    """Connect to switch via console device - TARGET is line number or device name of switch"""
+    lineNumber = ""
+    if devicename:
+        lineNumber = getLineNumber(target)
+        if lineNumber == "":
+            click.echo("Device {} does not exist".format(target))
+            return
+    else:
+        lineNumber = target
+    checkDevice(lineNumber)
+    lineNumber = str(lineNumber)
+
+    # QUIET == True => picocom will not output any messages, and pexpect will wait for console
+    #                  switch login or command line to let user interact with shell
+    #        Downside: if console switch output ever does not match READY_MSG, program will think connection failed
+    # QUIET == False => picocom will output messages - welcome message is caught by pexpect, so successful
+    #                   connection will always lead to user interacting with shell
+    #         Downside: at end of session, picocom will print exit message, exposing picocom to user
+    QUIET = False
+
+    baud, flowBool = getConnectionInfo(lineNumber)
+    baud = "9600" if baud == "" else baud
+    flowCmd = "h" if flowBool else "n"
+    quietCmd = "-q" if QUIET else ""
+    cmd = "sudo picocom -b {} -f {} {} {}{}".format(baud, flowCmd, quietCmd, DEVICE_PREFIX, lineNumber)
+    proc = pexpect.spawn(cmd)
+    proc.send("\n")
+
+    if QUIET:
+        READY_MSG = r"([Ll]ogin:|[$>#])" # login prompt or command line prompt
+    else:
+        READY_MSG = "Terminal ready" # picocom ready message
+    BUSY_MSG = "Resource temporarily unavailable" # picocom busy message
+    TIMEOUT_SEC = 0.2
+
+    index = proc.expect([READY_MSG, BUSY_MSG, pexpect.EOF, pexpect.TIMEOUT], timeout=TIMEOUT_SEC)
+    if index == 0: # terminal ready
```

Comment:
I am wondering if TIMEOUT_SEC, DEF_BAUD, DEFAULT_PREFIX should be defined as class attributes and initialized in a init function? So that we know that these are our global choices and available to see/change at a same location. QUIET could fall into this category, I think.

And READY_MSG, BUSY_MSG actually are local variables might be better to follow local variable naming convention. like readyMsg and busyMsg instead?

Reply:
I like that idea - how would the structure of that look? What class would they be attributes of?